src/bp-activity/bp-activity-template.php

```diff
@@ -4322,3 +4327,142 @@ function bp_get_activity_show_filters( $context = '' ) {
 		 */
 		return apply_filters( 'bp_get_activity_show_filters', $output, $filters, $context );
 	}
+
+
+/**
+ * Are Activity likes supported by this site?
+ *
+ * @since 14.0.0
+ * @todo Build the function!
+ *
+ * @return boolean True if Activity likes are supported. False otherwise.
+ */
+function bp_activity_supports_likes() {
+	/*
+	 * This is where we'll need to check for:
+	 * current_theme_supports( 'buddypress', ['activity', ['likes'] ] );
+	 */
+	return true;
+}
+
+/**
+ * Return whether an activity is liked by a user.
+ *
+ * @since 14.0.0
+ *
+ * @param integer $user_id     The user ID. Optional.
+ *                             Defaults to the current user ID.
+ * @param integer $activity_id The Activity object. Optional.
+ *                             Defaults to 0.
+ * @return boolean True if the user liked the activity. False otherwise.
+ */
+function bp_activity_is_liked( $user_id = 0, $activity_id = 0 ) {
+	if ( ! $user_id ) {
+		$user_id = bp_loggedin_user_id();
+	}
+
+	if ( $activity_id ) {
+		$user_like = bp_activity_get(
+			array(
+				'type'             => 'activity_like',
+				'filter'           => array(
+					'user_id'    => $user_id,
+					'primary_id' => $activity_id,
+				),
+				'display_comments' => 'stream',
```

Comment:
Do we need to append the comments too? This could be expensive.

Reply:
`display_comments` should be renamed so that it works for other type of activity reactions. This is required because `activity_like` is always attached to a parent activity, so you need to get activity children: this is only happening when `'display_comments' => 'stream'` in BuddyPress .

Reply:
Btw, a bit above, there's a type filter so that only `activity_like` one are returned + this type does not support comments.